assets/js/components/Dashboard/TimePicker.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { ComponentType, useEffect, useRef, useState } from "react";
+import { Card, Form, Overlay, Popover } from "react-bootstrap";
+import _ from "lodash";
+import fp from "lodash/fp";
+import moment from "moment";
+import cx from "classnames";
+
+interface TimePickerProps {
+  selectedTime: string;
+  onChange: (time: string) => void;
+}
+
+const TimePicker: ComponentType<TimePickerProps> = ({
+  selectedTime,
+  onChange,
+}: TimePickerProps) => {
+  const timeMoment = moment(selectedTime, "h:mm A");
+  const [showOverlay, setShowOverlay] = useState(false);
+  const [selectedHour, setSelectedHour] = useState(timeMoment.format("hh"));
+  const [selectedMinute, setSelectedMinute] = useState(timeMoment.format("mm"));
+  const [selectedAmPm, setSelectedAmPm] = useState(timeMoment.format("A"));
+  const ref = useRef(null);
+
+  const formatState = () =>
+    moment(
+      `${selectedHour}:${selectedMinute} ${selectedAmPm}`,
+      "hh:mm A",
+    ).format("HH:mm");
+
+  useEffect(() => {
+    onChange(formatState());
+  }, [selectedHour, selectedMinute, selectedAmPm]);
```

Comment:
**suggestion:** this looks like `useMemo` might be handy here

```suggestion
  const formattedState = useMemo(() => 
    moment(
      `${selectedHour}:${selectedMinute} ${selectedAmPm}`,
      "hh:mm A",
    ).format("HH:mm"),
    [selectedHour, selectedMinute, selectedAmPm]
  );

  useEffect(() => {
    onChange(formattedState);
  }, [formattedState]);
```

you could then update the other `onChange(formatState())` call in this file as well.

Reply:
`useMemo` is mainly used to avoid re-running expensive computations multiple times. Since this formatting is trivial, it should be fine to just compute it inline without `useMemo`